src/directLine.ts

```diff
@@ -272,15 +277,31 @@ export class DirectLine implements IBotConnection {
     constructor(options: DirectLineOptions) {
         this.secret = options.secret;
         this.token = options.secret || options.token;
+        const useWebSockets:boolean = options.webSocket && typeof WebSocket !== 'undefined' && WebSocket !== undefined;
+
         if (options.domain)
             this.domain = options.domain;
         if (options.webSocket !== undefined)
             this.webSocket = options.webSocket;
-
+        if (options.conversationId !== undefined) {
```

Comment:
We tend to just use `if (!options.conversationId)` for checking against undefined, unless the value is a number.